### PR TITLE
[APINotes] Add a 'SwiftImportAsNonGeneric' entry for ObjC classes

### PR DESCRIPTION
**Pull request by jrose-apple**
If true, Swift will ignore any 'lightweight generics' parameters when importing the class, substituting any uses of them with their bounds. At least, it will once this lands and the Swift side gets implemented.

Finishes rdar://problem/28455962.

### CODE REVIEW

**Comment by jrose-apple** [comment]
Mostly needs a rubber stamp from Doug.

File: lib/APINotes/APINotesWriter.cpp [hunk 1]
```diff
@@ -584,8 +584,12 @@ namespace {
       emitCommonTypeInfo(out, info);
 
       uint8_t payload = 0;
+      if (auto swiftImportAsNonGeneric = info.getSwiftImportAsNonGeneric()) {
+        payload |= (0x01 << 1) | swiftImportAsNonGeneric.getValue();
+      }
+      payload <<= 3;
```
Comment (DougGregor) [inline lib/APINotes/APINotesWriter.cpp:590]: Why are we shifting by 3 instead of 2? There are two bits of data here...

**Review by DougGregor — APPROVED** [review]

File: lib/APINotes/APINotesWriter.cpp (hunk 1 above)
Comment (jrose-apple) [reply lib/APINotes/APINotesWriter.cpp:590]: This is making room for the *next* chunk of data, not moving out the *previous* data.

Comment (DougGregor) [reply lib/APINotes/APINotesWriter.cpp:590]: Aaah, right. Guess I should just rubber-stamp. Thanks!

**Pull request merged**